cmd/exporter/main.go

```diff
@@ -87,17 +81,30 @@ func export() error {
 		ArtifactsDir: artifactsDir,
 	}
 
-	factory, err := image.NewFactory(image.WithOutWriter(os.Stdout), image.WithEnvKeychain)
-	if err != nil {
-		return err
-	}
-
 	var stack metadata.StackMetadata
 	_, err = toml.DecodeFile(stackPath, &stack)
 	if err != nil {
 		outLog.Printf("no stack.toml found at path '%s', stack metadata will not be exported\n", stackPath)
 	}
 
+	if runImageRef == "" {
+		if stack.RunImage.Image == "" {
+			return cmd.FailCode(cmd.CodeInvalidArgs, "--run-image is required when stack.toml is not present in builder")
+		}
+		runImageRef = stack.RunImage.Image
```

Comment:
We want to select the best mirror (the one whose registry matches  `repoName`) instead of always using `stack.RunImage.Image`. `stack.RunImage.Image` can be the default if none of the mirrors have a matching registry.